nabs/streams.py

```diff
@@ -0,0 +1,73 @@
+"""Standard data streams for automated routines"""
+import numpy as np
+from bluesky.callbacks.stream import LiveDispatcher
+
+
+class AverageStream(LiveDispatcher):
+    """
+    Stream that averages data points together
+
+    As event documents are emitted from the RunEngine they are collected by
+    AverageStream, averaged and re-emitted as a secondary event stream. This
+    allows callbacks to subscribe to the average stream for more elegant
+    visualization and analysis.
+
+    The average stream can either be constructed with a specific number of
+    points to average or this information can be passed through in the start
+    document metadata. See the :meth:`.start` for more information.
+
+    Parameters
+    ----------
+    n : int, optional
+        Number of points to average together
+    """
+    def __init__(self, n=None):
+        self.n = n
+        self.last_event = None
+        self.raw_cache = list()
+        super().__init__()
```

Comment:
Normally we do this first in an init. Does it matter for `LiveDispatcher` subclasses to do this last? Asking for knowledge gain.

Reply:
Shouldn't matter. I'll add it to top for consistency